src/main/java/org/folio/rest/core/BaseApi.java

```diff
@@ -33,18 +42,45 @@ public Future<Response> buildErrorResponse(Throwable throwable) {
     if (throwable instanceof HttpException httpException) {
       return buildErrorResponse(httpException.getStatusCode(), httpException.getPayload());
     }
+    if (throwable instanceof CustomFieldValidationException customFieldValidationException) {
+      return buildErrorResponse(
+          422, Json.encode(customFieldValidationException.getErrors()), APPLICATION_JSON);
+    }
     if (StringUtils.isNotBlank(PgExceptionUtil.badRequestMessage(throwable))) {
       return buildErrorResponse(Response.Status.BAD_REQUEST.getStatusCode(), PgExceptionUtil.badRequestMessage(throwable));
     }
     return buildErrorResponse(INTERNAL_SERVER_ERROR.getStatusCode(), throwable.getMessage());
   }
 
   private Future<Response> buildErrorResponse(int code, String message) {
+    return buildErrorResponse(code, message, TEXT_PLAIN);
+  }
+
+  private Future<Response> buildErrorResponse(int code, String message, String contentType) {
     return Future.succeededFuture(Response.status(code)
-      .header(CONTENT_TYPE, TEXT_PLAIN)
+      .header(CONTENT_TYPE, contentType)
       .entity(message)
       .build());
   }
 
+  private Map<String, Object> getCustomFields(Object entity) {
+    return JsonObject.mapFrom(entity)
+      .getJsonObject("customFields", new JsonObject())
+      .getMap()
+      .entrySet()
+      .stream()
+      .collect(Collectors.toMap(Entry::getKey, Entry::getValue));
+  }
+
+  public Future<Void> validateCustomFields(
+    Context vertxContext, Map<String, String> okapiHeaders, Object entity) {
+    Map<String, Object> customFieldsMap = getCustomFields(entity);
+    if (!customFieldsMap.isEmpty()) {
```

Comment:
we can add info log here that we started validation of custom fields, it can help match exception that it relies with custom fields validaiton, and please check that exception with invalid custom field logged

Reply:
I understand the importance of thorough logging, but I'm not entirely convinced that adding an additional log message would provide a significant benefit.

The exception is named `CustomFieldValidationException`, indicating an error related to custom fields and validation. It is logged by the calling API methods, for example [here](https://github.com/folio-org/mod-orders-storage/pull/380/files#diff-d16050af5f2ff623e9f6459f8f6e3d126a10421ba6202067162d5e2032e061d7R75-R77). The response generated by `buildErrorResponse` provides detailed information about the validation error(s).

Considering the provided context and the existing logging mechanisms, do you still think it's necessary to include the additional log message?

Reply:
I've updated the logging in `OrderTemplatesAPI` to align with the rest of the APIs.